pkg/mqtt-paho/Makefile

```diff
@@ -0,0 +1,25 @@
+PKG_NAME = mqtt-paho
+PKG_URL = https://github.com/eclipse/paho.mqtt.embedded-c.git
+PKG_VERSION = 29ab2aa29c5e47794284376d7f8386cfd54c3eed
+PKG_LICENSE = EDL
+
+#gathering all c and hdr files
+MQTT_PAHO_SRC = $(shell ls -1 $(PKG_BUILDDIR)/MQTTClient-C/src/*.c)
+MQTT_PAHO_SRC += $(shell find $(PKG_BUILDDIR)/MQTTPacket/src/ -name '*.c')
+#adding RIOT common interface functions
+MQTT_PAHO_SRC += $(shell find $(CURDIR)/contrib/ -name '*.c')
+MQTT_PAHO_HDRS = $(shell ls -1 $(PKG_BUILDDIR)/MQTTClient-C/src/*.h)
+MQTT_PAHO_HDRS += $(shell ls -1 $(PKG_BUILDDIR)/MQTTPacket/src/*.h)
+
+.PHONY: all
+
+all: prepare Makefile.riot
+
+	$(Q)mkdir -p $(PKG_BUILDDIR)/library
```

Comment:
I think you can void all this logic of copying files. Look at the tensorflow-lite package. It's possible to select a subset of files for a given module when setting the NO_AUTO_SRC variable and then set the SRCS with the C files that need to be built.
That would simplify a lot the way the package is built.